File: abc/abc162/abc162_c.cs
// https://atcoder.jp/contests/abc162/submissions/11796944

using System;

namespace CSharp
{
    class Program
    {
        static void Main()
        {
            var K = int.Parse(Console.ReadLine());

            var result = 0L;
            for (int i = 1; i <= K; i++)
            {
                for (int j = 1; j <= K; j++)
                {
                    for (int k = 1; k <= K; k++)
                    {
                        result += Gcd(Gcd(i, j), k);
                    }
                }
            }
            Console.WriteLine(result);
        }

        private static long Gcd(long a, long b)
        {
            if (a < b)
            {
                return Gcd(b, a);
            }
            while (b != 0)
            {
                var r = a % b;
                a = b;
                b = r;
            }
            return a;
        }
    }
}

File: abc/abc162/abc162_c.fs
// https://atcoder.jp/contests/abc162/submissions/11859490

let K = stdin.ReadLine() |> int

let rec gcd a b =
    match (a, b) with
    | (x, 0) -> x
    | (0, y) -> y
    | (a, b) -> gcd b (a % b)

seq {
    for i in 1..K do
        for j in 1..K do
            for k in 1..K do
                yield gcd i (gcd j k)
}
|> Seq.sum
|> stdout.WriteLine
